### ISSUE TITLE
Resulting GDI is unbootable, cannot open in tools such as jc-gdrom-explorer.

### ISSUE THREAD

**Issue by Kneesnap** [RedRingRico/cim2gdi#2]
Hello,

I'm doing some data preservation for a game, and I have one CD which is labelled as "DC 2/10/00". On this CD is a single .cim file. cim2gdi appears to run successfully on this cim file. However, the output GDI file (& included tracks) do not load in either Redream, or jc-gdrom-explorer. I'd imagine it might be somewhat involved to figure this out. I can send you the CD image. Thanks for the help!

![jc-gdrom-explorer](https://i.imgur.com/XcCCDLO.png)

**Comment by Kneesnap** [comment]
Here's the output of the program.
[output.txt](https://github.com/RedRingRico/cim2gdi/files/9947450/output.txt)

**Comment by RedRingRico** [comment]
Hey, there.  From the output it all looks okay.  If you could send me the image, I should be able to run it through and track down what went wrong.

**Comment by Kneesnap** [comment]
Here's a link to the image: https://drive.google.com/file/d/1PmVjsF-xeferWPLKuN8gcMpRH_0NYhEt/view?usp=sharing

**Comment by Kneesnap** [comment]
Hi there! Just hoping for a status update on this. Thanks! =)

**Comment by Kneesnap** [comment]
Something interesting I notice is how the file size of track03.raw in the output is around 3x the size of the .cim file itself.

**Comment by Kneesnap** [comment]
I made a .CIM file with the official KATANA SDK which doesn't work with this tool. Might that be helpful since it could allow for experimentation, and we know the exact input used to make the file?

**Comment by RedRingRico** [comment]
Hey, there.  Sorry I haven't had much time to look into this.  I did open the CIM file and tried to mount it on my development kit, but it came back with an error.  The issue with my tool is that I've only tested it against full-fat CIM files and not the type which aren't padded.

I've been able to listen to a very distorted warning message, but I'm not convinced that it's 100% incorrect.  I'll try and find some time to investigate further.

![CIMError](https://user-images.githubusercontent.com/3076138/204601227-f1f8d88b-7364-42f4-9cd2-a1ebe833ce26.png)

**Comment by Kneesnap** [comment]
I have a working setup which lets me build a .CIM file with the Dreamcast SDK from real game files (which I can use modernish tools to build a working .CDI image from), which cannot be read with CIM Utility, or cim2gdi. Perhaps you'd like me to share this setup with you? If you have a discord, it'd be easiest for me to help send it and give instructions. It works on Windows 10, but should also work in a VM just fine in theory.

**Comment by Kneesnap** [comment]
I've got an update, and I figured out the option which can cause it to fail / not fail. This doesn't fully fix my problem, but it may be helpful for you.
In CDI CD Craft, under Build Settings, if you do NOT check this checkbox, the resulting CIM file does not work with cim2gdi.
However, if you DO check the checkbox, then it DOES work.

This does not fully fix my problem, since I have a .CIM file which I did not make but would like to either extract or convert to GDI.

![image](https://user-images.githubusercontent.com/6562913/204687650-5db487d0-31c4-4a22-8c68-1e5247852a95.png)

**Comment by RedRingRico** [comment]
Yeah, that's what I was referring to with the non-padded CIM files.  I initially created it this way because I wasn't sure how to pad out the tracks without it.  It's listed as one of the goals in the roadmap of the README.

It didn't occur to me that studios wouldn't use the full image format when creating their CIM files for testing, though it makes sense, given how much more it can add to the file.  I've been meaning to improve the tool, but with little motivation to do so as it fit my needs for all this time.

I'll make an effort to take the time to work on this, as it bothers me that it's of sub-par quality.

**Comment by Kneesnap** [comment]
Sounds good!

**Comment by Kneesnap** [comment]
Just checking in on this! What's the plan for this? I'm hoping to be able to release this prototype at some point, so it'd be nice to have an idea of what to expect in terms of timeframe. Don't mean to pressure you though!

**Comment by RedRingRico** [comment]
No problem.  I'll be able to get back to this over the weekend.  Another long-standing issue is to move over to using FASTBuild instead of relying on Visual Studio to build the project.  The Windows machine this was originally developed on still exists, so I'll concentrate on getting the non-padded image working before moving it over.

**Comment by RedRingRico** [comment]
Almost there.  It can be open in the GD-ROM Explorer but fails on the console.  At least it blanks the screen, but there's no other activity.  I had to manually intervene to get track04.raw to the correct size.  I think it's most likely junk info in there as the contents don't look right.

![image](https://user-images.githubusercontent.com/3076138/208310265-149e146e-2230-466d-9484-45309e7ea3c4.png)

**Comment by Kneesnap** [comment]
Nice! Looking forward to getting this worked out!

**Comment by Kneesnap** [comment]
> Almost there. It can be open in the GD-ROM Explorer but fails on the console. At least it blanks the screen, but there's no other activity. I had to manually intervene to get track04.raw to the correct size. I think it's most likely junk info in there as the contents don't look right.
> 
> ![image](https://user-images.githubusercontent.com/3076138/208310265-149e146e-2230-466d-9484-45309e7ea3c4.png)

Does this match track04?
[chickenrun.zip](https://github.com/RedRingRico/cim2gdi/files/10306554/chickenrun.zip)
I have a backup of the final CD image unbuilt and this is track4 there. It's possible this changed since the prototype though.

**Comment by Kneesnap** [comment]
Just checking in on the status of this since it's been more than a month. I assume you were probably busy with the holidays, no worries.

**Comment by Kneesnap** [comment]
what's the status here? It'd be nice to have some kind of update

**Comment by Kneesnap** [comment]
Hey, I hope this isn't a bother, but having this functionality soon will let me release a prototype.
